internals/overlord/logstate/syslog/syslog_test.go

```diff
@@ -287,3 +290,91 @@ func (s *testSyslogServer) run() error {
 
 	return nil
 }
+
+func (*suite) TestUDPAddAndFlush(c *C) {
+	conn, err := net.ListenPacket("udp", "localhost:0")
+	c.Assert(err, IsNil)
+	defer conn.Close()
+
+	msgChan := make(chan string, 1)
+	go func() {
+		for {
+			buf := make([]byte, 8192)
+			n, _, err := conn.ReadFrom(buf)
+			if err != nil {
+				c.Assert(errors.Is(err, net.ErrClosed), Equals, true)
+				break
+			}
+			if n > 0 {
+				msgChan <- string(buf[:n])
+			}
+		}
+		close(msgChan)
+	}()
+
+	client, err := syslog.NewClient(&syslog.ClientOptions{
+		Location: "udp://" + conn.LocalAddr().String(),
+		Hostname: "test-machine",
+		SDID:     "test-sdid",
+	})
+	c.Assert(err, IsNil)
+	defer client.Close()
+
+	client.SetLabels("svc1", map[string]string{
+		"env":     "test",
+		"version": "0.0.1",
+	})
+	for _, entry := range exampleEntries {
+		err = client.Add(entry)
+		c.Assert(err, IsNil)
+	}
+
+	err = client.Flush(context.Background())
+	c.Assert(err, IsNil)
+
+	expectedLogMsg := []string{
+		`<13>1 2023-12-31T12:00:00.123456789Z test-machine svc1 - - [test-sdid@28978 env="test" version="0.0.1"] message from svc1`,
+		`<13>1 2023-12-31T12:00:01.123456789Z test-machine svc2 - - - msg from svc2`,
+		`<13>1 2023-12-31T12:00:02.123456789Z test-machine svc1 - - [test-sdid@28978 env="test" version="0.0.1"] long message from svc1`,
+		`<13>1 2023-12-31T12:00:03.123456789Z test-machine svc3 - - - log of svc3 doesn't have any labels`,
+		`<13>1 2023-12-31T12:00:04.123456789Z test-machine svc4 - - - multiline
+line2
+line3`,
+	}
+	for _, expected := range expectedLogMsg {
+		select {
+		case msg := <-msgChan:
+			c.Check(msg, Equals, expected)
+		case <-time.After(2 * time.Second):
+			c.Fatal("timed out waiting for UDP message")
+		}
+	}
+
+	// Test message longer than max UDP size is truncated
+	longMsg := make([]byte, 5000)
+	for i := range longMsg {
+		longMsg[i] = 'A'
+	}
+
+	c.Assert(client.Add(servicelog.Entry{
+		Time:    time.Date(2023, 12, 31, 12, 0, 0, 0, time.UTC),
+		Service: "svc2",
+		Message: string(longMsg),
+	}), IsNil)
+	c.Assert(client.Flush(context.Background()), IsNil)
+
+	select {
+	case shortMsg := <-msgChan:
+		c.Check(len(longMsg) == 5000, Equals, true)
```

Comment:
I don't see any need for this assertion. It's guaranteed to be 5000 on creation.